lib/dashx/client.rb

```diff
@@ -0,0 +1,60 @@
+require 'httparty'
+
+module DashX
+  class Client
+    include HTTParty
+    base_uri 'https://api.dashx.com/v1'
+
+    def initialize(config)
+      @config = config
+
+      self.class.base_uri(config[:base_uri])
+      self.class.headers({
+        'X-Public-Key' => config[:public_key],
+        'X-Private-Key' => config[:private_key]
+      })
+    end
+
+    def make_http_request(uri, body)
```

Comment:
Place `make_http_request`, `symbolize_keys!` and `check_presence!` in a private block at the bottom of the file.